src/main/haskell/kore/src/Kore/IndexedModule/MetadataTools.hs

```diff
@@ -58,21 +66,39 @@ extractMetadataTools m =
     , sortAttributes = getSortAttributes m
     , symbolOrAliasSorts  = getHeadApplicationSorts m
     , isSubsortOf = checkSubsort
+    , subsorts = Set.fromList . fmap getSortFromId . getSubsorts
     }
   where
+    subsortTable :: Map (Sort Object) [Sort Object]
```

Comment:
You should add unit-tests for this.

Reply:
Done; see https://github.com/kframework/kore/pull/298/commits/2ffd4b051b9fa1c141e1ba7d52e9555f049fcfbc#diff-c1602aa6b3c4f86ae4b16a7770174399R193 (and below)